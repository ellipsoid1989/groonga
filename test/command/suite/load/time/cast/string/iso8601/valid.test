plugin_register functions/time
table_create Logs TABLE_NO_KEY
column_create Logs case COLUMN_SCALAR ShortText
column_create Logs created_at COLUMN_SCALAR Time
column_create Logs created_at_text COLUMN_SCALAR ShortText

load --table Logs
[
{"case": "timezone: Z", "created_at": "2000-01-01T10:00:00Z", "created_at_text": "2000-01-01T10:00:00Z"},
{"case": "timezone: z", "created_at": "2000-01-01t10:00:00z", "created_at_text": "2000-01-01T10:00:00z"},
{"case": "timezone: 00:00", "created_at": "2000-01-01T10:00:00+00:00", "created_at_text": "2000-01-01T10:00:00+00:00"},
{"case": "timezone: +01:01", "created_at": "2000-01-01T11:01:00+01:01", "created_at_text": "2000-01-01T11:01:00+01:01"},
{"case": "timezone: +11:11", "created_at": "2000-01-01T21:11:00+11:11", "created_at_text": "2000-01-01T21:11:00+11:11"},
{"case": "timezone: -01:01", "created_at": "2000-01-01T08:59:00-01:01", "created_at_text": "2000-01-01T08:59:00-01:01"},
{"case": "timezone: -11:11", "created_at": "1999-12-31T22:49:00-11:11", "created_at_text": "1999-12-31T22:49:00-11:11"},
{"case": "timezone hour threshold: +23:00", "created_at": "2000-01-02T09:00:00+23:00", "created_at_text": "2000-01-02T09:00:00+23:00"},
{"case": "timezone minute threshold: +00:59", "created_at": "2000-01-01T10:59:00+00:59", "created_at_text": "2000-01-01T10:59:00+00:59"},
{"case": "timezone omitting minute: +01", "created_at": "2000-01-01T11:00:00+01", "created_at_text": "2000-01-01T11:00:00+01"},
{"case": "timezone omitting minute: -01", "created_at": "2000-01-01T09:00:00-01", "created_at_text": "2000-01-01T09:00:00-01"},
{"case": "timezone: localtime", "created_at": "2000-01-01T19:00:00", "created_at_text": "2000-01-01T19:00:00"},
{"case": "compatible: date delimiter: /", "created_at": "2000/01/01T10:00:00Z", "created_at_text": "2000/01/01T10:00:00Z"},
{"case": "decimal", "created_at": "2000-01-01T11:01:00.123+01:01", "created_at_text": "2000-01-01T11:01:00.123+01:01"}
]

select Logs --limit -1 --output_columns "case, time_format_iso8601(created_at), created_at_text"
