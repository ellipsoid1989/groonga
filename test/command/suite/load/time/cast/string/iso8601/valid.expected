plugin_register functions/time
[[0,0.0,0.0],true]
table_create Logs TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Logs case COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Logs created_at COLUMN_SCALAR Time
[[0,0.0,0.0],true]
column_create Logs created_at_text COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
load --table Logs
[
{"case": "timezone: Z", "created_at": "2000-01-01T10:00:00Z", "created_at_text": "2000-01-01T10:00:00Z"},
{"case": "timezone: z", "created_at": "2000-01-01t10:00:00z", "created_at_text": "2000-01-01T10:00:00z"},
{"case": "timezone: 00:00", "created_at": "2000-01-01T10:00:00+00:00", "created_at_text": "2000-01-01T10:00:00+00:00"},
{"case": "timezone: +01:01", "created_at": "2000-01-01T11:01:00+01:01", "created_at_text": "2000-01-01T11:01:00+01:01"},
{"case": "timezone: +11:11", "created_at": "2000-01-01T21:11:00+11:11", "created_at_text": "2000-01-01T21:11:00+11:11"},
{"case": "timezone: -01:01", "created_at": "2000-01-01T08:59:00-01:01", "created_at_text": "2000-01-01T08:59:00-01:01"},
{"case": "timezone: -11:11", "created_at": "1999-12-31T22:49:00-11:11", "created_at_text": "1999-12-31T22:49:00-11:11"},
{"case": "timezone hour threshold: +23:00", "created_at": "2000-01-02T09:00:00+23:00", "created_at_text": "2000-01-02T09:00:00+23:00"},
{"case": "timezone minute threshold: +00:59", "created_at": "2000-01-01T10:59:00+00:59", "created_at_text": "2000-01-01T10:59:00+00:59"},
{"case": "timezone omitting minute: +01", "created_at": "2000-01-01T11:00:00+01", "created_at_text": "2000-01-01T11:00:00+01"},
{"case": "timezone omitting minute: -01", "created_at": "2000-01-01T09:00:00-01", "created_at_text": "2000-01-01T09:00:00-01"},
{"case": "timezone: localtime", "created_at": "2000-01-01T19:00:00", "created_at_text": "2000-01-01T19:00:00"},
{"case": "compatible: date delimiter: /", "created_at": "2000/01/01T10:00:00Z", "created_at_text": "2000/01/01T10:00:00Z"},
{"case": "decimal", "created_at": "2000-01-01T11:01:00.123+01:01", "created_at_text": "2000-01-01T11:01:00.123+01:01"}
]
[[0,0.0,0.0],14]
select Logs --limit -1 --output_columns "case, time_format_iso8601(created_at), created_at_text"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        14
      ],
      [
        [
          "case",
          "ShortText"
        ],
        [
          "time_format_iso8601",
          null
        ],
        [
          "created_at_text",
          "ShortText"
        ]
      ],
      [
        "timezone: Z",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T10:00:00Z"
      ],
      [
        "timezone: z",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T10:00:00z"
      ],
      [
        "timezone: 00:00",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T10:00:00+00:00"
      ],
      [
        "timezone: +01:01",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T11:01:00+01:01"
      ],
      [
        "timezone: +11:11",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T21:11:00+11:11"
      ],
      [
        "timezone: -01:01",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T08:59:00-01:01"
      ],
      [
        "timezone: -11:11",
        "2000-01-01T19:00:00.000000+09:00",
        "1999-12-31T22:49:00-11:11"
      ],
      [
        "timezone hour threshold: +23:00",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-02T09:00:00+23:00"
      ],
      [
        "timezone minute threshold: +00:59",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T10:59:00+00:59"
      ],
      [
        "timezone omitting minute: +01",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T11:00:00+01"
      ],
      [
        "timezone omitting minute: -01",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T09:00:00-01"
      ],
      [
        "timezone: localtime",
        "2000-01-01T19:00:00.000000+09:00",
        "2000-01-01T19:00:00"
      ],
      [
        "date separater: /",
        "2000-01-01T19:00:00.000000+09:00",
        "2000/01/01T10:00:00Z"
      ],
      [
        "decimal",
        "2000-01-01T19:00:00.123000+09:00",
        "2000-01-01T11:01:00.123+01:01"
      ]
    ]
  ]
]
